### PR TITLE
CLOUDP-167793:Migrate Atlas Search CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167793](https://jira.mongodb.org/browse/CLOUDP-167793)

EVG Patch: https://spruce.mongodb.com/version/6421cc400ae606fcdcda3b85/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/search/index_opts.go [hunk 1]
```diff
@@ -86,28 +86,28 @@ func (opts *IndexOpts) newSearchIndex() (*atlas.SearchIndex, error) {
...
-		Mappings: &atlas.IndexMapping{
-			Dynamic: opts.dynamic,
-			Fields:  &f,
+		Mappings: &atlasv2.FTSMappings{
+			Dynamic: &opts.dynamic,
+			Fields:  f,
 		},
 		Name:           opts.name,
-		SearchAnalyzer: opts.searchAnalyzer,
+		SearchAnalyzer: &opts.searchAnalyzer,
```
Comment (matt-condon) [inline internal/cli/atlas/search/index_opts.go:98]: 🥇 
I like this approach
Let's try to amend the input model like this where possible for the "Atlas only" commands @swattyT @andmatei

**Review by matt-condon — DISMISSED** [review]

File: internal/store/search.go [hunk 2]
```diff
@@ -19,68 +19,69 @@ import (
...
+	UpdateSearchIndexes(string, string, string, *atlasv2.FTSIndex) (*atlasv2.FTSIndex, error)
 }
 
 type SearchIndexDeleter interface {
 	DeleteSearchIndex(string, string, string) error
 }
 
 // SearchIndexes encapsulate the logic to manage different cloud providers.
-func (s *Store) SearchIndexes(projectID, clusterName, dbName, collName string, opts *atlas.ListOptions) ([]*atlas.SearchIndex, error) {
+func (s *Store) SearchIndexes(projectID, clusterName, dbName, collName string, _ *atlas.ListOptions) ([]atlasv2.FTSIndex, error) {
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/store/search.go:48]: let's delete it if unused

**Review by gssbzn — COMMENTED** [review]
Search is a complicated piece, can we make sure some of the examples described at https://www.mongodb.com/docs/atlas/atlas-search/define-field-mappings/#examples still work, maybe even add them as e2e test as the current test we have is a bit basic

**Comment by wtrocki** [comment]
> Search is a complicated piece, can we make sure some of the examples described at https://www.mongodb.com/docs/atlas/atlas-search/define-field-mappings/#examples still work, maybe even add them as e2e test as the current test we have is a bit basic

@swattyT to improve visibility maybe we can create jira for e2e tests.

File: test/e2e/atlas/search_test.go [hunk 3]
```diff
@@ -303,7 +303,7 @@ func TestSearch(t *testing.T) {
 		if err != nil {
 			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
 		}
-		var index mongodbatlas.SearchIndex
+		var index atlasv2.FTSIndex
```
Comment (wtrocki) [inline test/e2e/atlas/search_test.go:306]: I have flagged this for possible rename for GA epic.

File: internal/cli/atlas/search/create_test.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ import (
 	"github.com/golang/mock/gomock"
 	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
 	"github.com/spf13/afero"
-	"go.mongodb.org/atlas/mongodbatlas"
+	atasv2 "go.mongodb.org/atlas/mongodbatlasv2"
```
Comment (wtrocki) [inline internal/cli/atlas/search/create_test.go:25]: image

**Review by wtrocki — APPROVED** [review]
Awesome work!

**Pull request merged**